cscs-checks/libraries/math/trilinos_compile_run.py

```diff
@@ -8,48 +8,52 @@
 import reframe.utility.sanity as sn
 
 
-@rfm.parameterized_test(['static'], ['dynamic'])
+@rfm.simple_test
 class TrilinosTest(rfm.RegressionTest):
-    def __init__(self, linkage):
-        self.valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc']
+    linkage = parameter(['static', 'dynamic'])
+    valid_systems = ['daint:gpu', 'daint:mc', 'dom:gpu', 'dom:mc']
+    valid_prog_environs = ['PrgEnv-gnu', 'PrgEnv-intel']
+    prgenv_flags = {
+        'PrgEnv-cray': ['-fopenmp', '-O2', '-ffast-math', '-std=c++11',
+                        '-Wno-everything'],
+        'PrgEnv-cray_classic': ['-homp', '-hstd=c++11', '-hmsglevel_4'],
+        'PrgEnv-gnu': ['-fopenmp', '-std=c++11', '-w', '-fpermissive'],
+        'PrgEnv-intel': ['-qopenmp', '-w', '-std=c++11'],
+        'PrgEnv-pgi': ['-mp', '-w']
+    }
+    sourcepath = 'example_AmesosFactory_HB.cpp'
+    prerun_cmds = ['wget ftp://math.nist.gov/pub/MatrixMarket2/'
+                   'misc/hamm/add20.rua.gz', 'gunzip add20.rua.gz']
+    executable_opts = ['add20.rua']
+    modules = ['cray-mpich', 'cray-hdf5-parallel', 'cray-tpsl',
+               'cray-trilinos']
+    num_tasks = 2
+    num_tasks_per_node = 2
+    variables = {'OMP_NUM_THREADS': '1'}
```

Comment:
I would put this in a pre-run hook, since it's only related to the run phase of the test.